torch/nn/modules/container.py

```diff
@@ -27,21 +27,50 @@ def __init__(self, **kwargs: Any) -> None:
 
 
 class Sequential(Module):
-    r"""A sequential container.
-    Modules will be added to it in the order they are passed in the constructor.
-    Alternatively, an ordered dict of modules can also be passed in.
+    r"""A list of ``Module``s that acts as a ``Module`` itself.
+    A ``Sequential`` is fundamentally a list of ``Module``s, each with a 
+    ``forward()`` method. ``Sequential`` provides a ``forward()`` method
+    of its own, which accepts any input and forwards it to the first 
+    module it stores. It then "chains" outputs to inputs sequentially 
+    for each subsequent module, finally returning the output of the last
+    module. Modules will be added to it in the order they are passed in 
+    the constructor. Alternatively, an ``OrderedDict`` of modules can 
+    be passed in.
+
+    Why should you use a ``Sequential`` instead of a ``ModuleList``? 
+    The value a ``Sequential`` provides over manually calling a sequence
+    of modules is that it allows treating the whole container as a 
+    single module, such that performing a transformation on the 
+    ``Sequential`` applies to each of the modules it stores (which are 
+    each a registered submodule of the ``Sequential``). Also, unlike
+    ``Sequential``,  a ``ModuleList`` does not have a ``forward()`` 
+    method, because it does not define a neural network; in fact, there 
+    is no connection between each of the Modules that it stores. A
+    ``ModuleList`` is exactly what it sounds like--a list for storing
+    ``Module``s! On the other hand, the layers in a ``Sequential`` are 
+    connected in a cascading way.
+
+    If you're using ``Sequential`` with TorchScript, the inputs of some
+    of the ``Sequential`` submodules may be falsely inferred to be
+    ``Tensor``, even if they're annotated otherwise. This is a known
+    issue in the TorchScript type inference algorithm. The canonical
+    solution is to subclass ``nn.Sequential`` and redeclare ``forward`` 
+    with the input typed correctly.
 
-    To make it easier to understand, here is a small example::
+    Example::
 
-        # Example of using Sequential
+        # Using Sequential to create a small model. When `model` is run,
+        # input will first be passed to `Conv2d(1,20,5)`, then to 
+        # `ReLU`, then to `Conv2d(20,64,5)`, then to `ReLU` again
```

Comment:
This wording is slightly confusing to me; it sounds like the same initial input will be passed to each item in the list. I think the description should better match the chaining behavior you described well in the first paragraph.

Also, tiny nit: "then to `ReLU` again" -> "then to the second `ReLU`" or something to be unambiguous about the linear chaining behavior.